### PR TITLE
Add RBIs for missing definitions in stdlib

### PR DESCRIPTION
**Pull request by Morriar**
We have a few more definitions for stdlib that do not have signatures but I think it still can be useful as:
* The constants and methods are defined which allow more code to be `typed: true`
* The documentation can popup in the editor

Here's the list of modules included in this PR:
* `dbm`
* `fiddle`
* `gdbm`
* `getoptlong`
* `nfk`
* `obersable`
* `readline`
* `openuri`
* `prime`
* `pstore`
* `rinda`
* `shellwords`
* `thwait`
* `weakref`

### CODE REVIEW

**Comment by jez** [comment]
I'm seeing the same errors you see on buildkite on Stripe's codebase. Do you mind taking a look? It looks like the current PR causes Sorbet's own payload RBIs to not type check.

**Comment by Morriar** [comment]
This one needed to be rebased on master to get `BRb`.

**Comment by Morriar** [comment]
Updated exp files with `tools/scripts/update_exp_files.rb`

File: gems/sorbet/test/hidden-method-finder/simple/ruby_2_5_hidden.rbi.exp [hunk 1]
```diff
@@ -9842,3 +9842,4 @@ end
 module Warning
   extend ::Warning
 end
+
```
Comment (Morriar) [inline gems/sorbet/test/hidden-method-finder/simple/ruby_2_5_hidden.rbi.exp:9845]: Not sure why this is now needed?

**Comment by jez** [comment]
https://go/builds/bui_HJZRPaaMb9Bt0d

File: rbi/stdlib/weakref.rbi [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+#
+# ```ruby
+# foo = Object.new            # create a new object instance
+# p foo.to_s                  # original's class
+# foo = WeakRef.new(foo)      # reassign foo with WeakRef instance
+# p foo.to_s                  # should be same class
+# GC.start                    # start the garbage collector
+# p foo.to_s                  # should raise exception (recycled)
+# ```
+class WeakRef
```
Comment (jez) [inline rbi/stdlib/weakref.rbi:18]: When I tested this on Stripe's CI, it said that `WeakRef` descends from `Delegator`, which i confirmed via IRB:

```
❯ irb
irb(main):001:0> require 'weakref'
=> true
irb(main):002:0> WeakRef.ancestors
=> [WeakRef, Delegator, #<Module:0x00007fcfee88a8e0>, BasicObject]
```

```suggestion
class WeakRef < Delegator
```

File: gems/sorbet/test/hidden-method-finder/simple/ruby_2_5_hidden.rbi.exp (hunk 1 above)
Comment (jez) [reply gems/sorbet/test/hidden-method-finder/simple/ruby_2_5_hidden.rbi.exp:9845]: Me neither.

File: rbi/stdlib/weakref.rbi (hunk 2 above)
Comment (Morriar) [reply rbi/stdlib/weakref.rbi]: Good catch. I added the superclass.

**Comment by jez** [comment]
These changes were included via #3043

**Pull request closed without merging**